skfp/descriptors/topological.py

```diff
@@ -288,6 +330,156 @@ def polarity_number(
     return int((distance_matrix == 3).sum() // 2)
 
 
+def kappa1_index(mol: Mol) -> float:
+    r"""
+    First Kappa shape index (K1).
+
+    Computes the first kappa shape index [1]_, which measures molecular shape based on
+    single bonds. It is given by the equation:
+
+    .. math::
+        K_1 = \frac{(A + \alpha) (A + \alpha - 1)^2}{P_1^2}
+
+    where:
+    - A is the number of heavy atoms,
+    - α is the Hall-Kier alpha index,
+    - P1 is the number of single bonds.
+
+    This index provides insight into the molecular shape and branching properties.
+
+    Parameters
+    ----------
+    mol : RDKit ``Mol`` object
+        The molecule for which the first Kappa shape index is calculated.
+
+    References
+    ----------
+    .. [1] `Milan, Randić
+        "Novel Shape Descriptors for Molecular Graphs"
+        Journal of Chemical Information and Computer Sciences 41.3 (2001):  607–613.
+        <https://pubs.acs.org/doi/10.1021/ci0001031>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.topological import kappa1_index
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> kappa1_index(mol)
+    0.0
+    """
+    P1 = bond_count(mol, "SINGLE")
+    A = heavy_atom_count(mol)
+    alpha = float(hall_kier_alpha(mol))
+
+    if P1 == 0:
+        return 0.0
+    return ((A + alpha) * (A + alpha - 1) ** 2) / (P1**2)
```

Comment:
You have all 3 kappa indices implemented in RDKit: https://www.rdkit.org/docs/source/rdkit.Chem.GraphDescriptors.html#rdkit.Chem.GraphDescriptors.Kappa1. Simply call those functions